device/src/async_device/test/mod.rs

```diff
@@ -34,9 +34,41 @@ async fn test_join_rx1() {
 
     // Await the device to return and verify state
     if let Ok(()) = async_device.await.unwrap() {
-        // NB: timer is armed twice (even if not fired twice)
-        // because RX1 end is armed when packet is received
-        assert_eq!(2, timer.get_armed_count().await);
+        // NB: timer is armed thrice (even if not fired thrice)
+        // 1. start of rx1
+        // 2. timeout after preamble (not firing)
+        // 3. end of rx1
+        assert_eq!(3, timer.get_armed_count().await);
+    } else {
+        panic!();
+    }
+}
+
+#[ignore]
+#[tokio::test]
+async fn test_join_long_rx1() {
+    let (radio, timer, mut async_device) = setup();
+    // Run the device
+    let async_device =
+        tokio::spawn(async move { async_device.join(&get_otaa_credentials()).await });
+
+    // Trigger beginning of RX1
+    timer.fire().await;
+    // RX preamble
+    radio.fire_preamble();
+    // Preamble received
+    timer.fire().await;
+
+    // Trigger handling of JoinAccept
+    radio.handle_rxtx(handle_join_request);
```

Comment:
I'm a little confused that this doesn't trigger the panic from `Msg::Preamble, TargetRxState::Done` condition. The fire_preamble should have but us targeting the `TargetRxState::Done` event and handle_rxtx precedes the frame with another preamble... I'll take a look later but wanted to drop the note.